urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/examples/QueueStackState.java

```diff
@@ -0,0 +1,117 @@
+package permlab.examples;
+
+import java.util.ArrayDeque;
+import java.util.HashSet;
+import permlib.classes.PermutationClass;
+import permlib.Permutation;
+import permlib.Permutations;
+import permlib.processor.PermProcessor;
+
+/**
+ *
+ * @author MichaelAlbert
+ */
+public class QueueStackState {
+
+    ArrayDeque<Integer> q = new ArrayDeque<Integer>();
+    ArrayDeque<Integer> s = new ArrayDeque<Integer>();
+    int nextOut = 0;
+
+    public QueueStackState() {
+    }
+
+    ;
+    public QueueStackState(QueueStackState other) {
+        q = other.q.clone();
+        s = other.s.clone();
+        nextOut = other.nextOut;
+    }
+
+    public QueueStackState clone() {
+        return new QueueStackState(this);
+    }
+
+    public boolean sortable(int[] input) {
+        ArrayDeque<Integer> in = new ArrayDeque<Integer>();
+        for (int i : input) {
+            in.addLast(i);
+        }
+        return sortable(in);
+    }
+
+    public boolean sortable(Permutation p) {
+        return sortable(p.elements);
+    }
+
+    public boolean sortable(ArrayDeque<Integer> input) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 57-86 into a separate method.